src/Microsoft.AspNetCore.Razor.Language/Syntax/MarkupElementRewriter.cs

```diff
@@ -69,9 +69,7 @@ private SyntaxNode RewriteNode(SyntaxNode node)
                     }
 
                     var tagName = tagBlock.GetTagName();
-                    if (string.IsNullOrWhiteSpace(tagName) ||
-                        tagBlock.IsVoidElement() ||
```

Comment:
No longer special casing Void elements at this level (Just moved this logic to the tag helper phase).
We can always add the special casing later if we end up needing this for non-tag helper scenarios (Like Blazor).